tests/subsys/net/lib/nrf_cloud/cloud/src/main.c

```diff
@@ -0,0 +1,1880 @@
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Init the cloud successfully */
+	init_cloud_success();
+
+	/* Connect to the cloud successfully with dc_connect */
+	connect_cloud_dc_success();
+
+	nrf_cloud_encode_sensor_data_fake.custom_fake = fake_nrf_cloud_encode_sensor_data__fails;
+
+	int ret = nrf_cloud_sensor_data_stream(&sensor_param_t);
+
+	zassert_not_equal(0, ret, "return should not be 0 when sensor data encoding fails");
+}
+
+/* Verify that the function will fail if the dc_stream function fails */
+ZTEST(nrf_cloud_test, test_cloud_sensor_data_stream_dc_stream_failed)
+{
+	/* Sensor data for sending to cloud */
+	struct nrf_cloud_data cloud_data_t = {
+		.len = 0U,
+		.ptr = NULL,
+	};
+
+	struct nrf_cloud_sensor_data sensor_param_t = {
+		.type = NRF_CLOUD_SENSOR_GPS,
+		.data = cloud_data_t,
+		.tag = 20000U,
+	};
+
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Init the cloud successfully */
+	init_cloud_success();
+
+	/* Connect to the cloud successfully with dc_connect */
+	connect_cloud_dc_success();
+
+	nrf_cloud_encode_sensor_data_fake.custom_fake = fake_nrf_cloud_encode_sensor_data__succeeds;
+	nct_dc_stream_fake.custom_fake = fake_nct_dc_stream__fails;
+
+	int ret = nrf_cloud_sensor_data_stream(&sensor_param_t);
+
+	zassert_not_equal(0, ret, "return should not be 0 when sensor data streaming fails");
+}
+
+/* Verify that the function will succeed given the
+ * suitable state, data encoding and streaming succeed
+ */
+ZTEST(nrf_cloud_test, test_cloud_sensor_data_stream_success)
+{
+	/* Sensor data for sending to cloud */
+	struct nrf_cloud_data cloud_data_t = {
+		.len = 0U,
+		.ptr = NULL,
+	};
+
+	struct nrf_cloud_sensor_data sensor_param_t = {
+		.type = NRF_CLOUD_SENSOR_GPS,
+		.data = cloud_data_t,
+		.tag = 20000U,
+	};
+
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Init the cloud successfully */
+	init_cloud_success();
+
+	/* Connect to the cloud successfully with dc_connect */
+	connect_cloud_dc_success();
+
+	nrf_cloud_encode_sensor_data_fake.custom_fake = fake_nrf_cloud_encode_sensor_data__succeeds;
+	nct_dc_stream_fake.custom_fake = fake_nct_dc_stream__succeeds;
+
+	int ret = nrf_cloud_sensor_data_stream(&sensor_param_t);
+
+	zassert_ok(ret, "return should be 0 when success");
+}
+
+/* TEST NRF_CLOUD_SEND */
+
+/* Verify that the function will fail
+ * if the message to be sent is NULL
+ */
+ZTEST(nrf_cloud_test, test_cloud_send_null_message)
+{
+	int ret = nrf_cloud_send(NULL);
+
+	zassert_equal(-EINVAL, ret, "return should be -EINVAL with NULL message");
+}
+
+/* Verify that the function will fail if
+ * the cloud is not in cc_connected state
+ */
+ZTEST(nrf_cloud_test, test_cloud_send_not_cc_connected)
+{
+	/* Message data */
+	struct nrf_cloud_tx_data tx_data_t = {
```

Comment:
same for these...
you could have a few:
```
const struct nrf_cloud_tx_data tx_msg_qos1 = {
		.topic_type = NRF_CLOUD_TOPIC_MESSAGE,
		.qos = MQTT_QOS_1_AT_LEAST_ONCE
	};
```
```
const struct nrf_cloud_tx_data tx_state_qos1 = {
		.topic_type = NRF_CLOUD_TOPIC_STATE,
		.qos = MQTT_QOS_1_AT_LEAST_ONCE
	};
```

... just to eliminate lines of code.

Reply:
I only have 2-3 of these, so I would rename them as you said but won't make them global, since I change those `topic_type` and `qos` several times. Thanks